File: lab6.cpp
#include <iostream>
#include <cmath>
using namespace std;


double result = 0;
double sumEquation(int n) {
    if (n > 0) {
        result += pow(-1, n+1) / n;
        sumEquation(n-1);
    }
    return result;
}

double sumEquation() {
    result = 0;
    int n;
    while (true){
        cout <<"Enter the n value: ";
        cin >> n;
        if (n < 1) {
            cout <<"Please enter a number that is greater or equal to 1" << endl;
        }else{
            break;
        }
    }

    result = sumEquation(n);
    return result;
}

int main() {
    int n;
    while (true){
        cout <<"Enter the n value: ";
        cin >> n;
        if (n < 1) {
            cout <<"Please enter a number that is greater or equal to 1" << endl;
        }else{
            break;
        }
    }
    cout <<"Result of the question is: " << sumEquation(n) << endl;

    double answer = sumEquation();
    cout <<"Result of the question is: " << answer << endl;

    return 0;
}

File: lab6.py
import math
#Question1

n = int(input("Please enter the n value: "))
while True:
    x = int(input("Please enter the x value: "))
    if x < 0:
        print("X value has to be greater than 0")
    else:
        break
equation = sum([(lambda i : n**i/math.factorial(i))(i) for i in range(x+1)])
print(equation)

#Question2

result = 0
def sumEquation(n): #This function calculates the summation of the series which is given in lab6 question2
    global result
    #recursion will work until n is not greater than 0
    if n > 0:
        result += (-1) ** (n + 1) / n
        sumEquation(n - 1) # recursion


while True:
    n2 = int(input("Enter the n value: "))
    if n2 < 1:
        print("Please enter a number that is greater or equal to 1")
    else:
        break;
sumEquation(n2)

print("Result of the second question is:", result)
